Microsoft.Toolkit.Uwp.UI.Controls/InAppNotification/InAppNotification.cs

```diff
@@ -67,16 +72,20 @@ protected override void OnApplyTemplate()
         /// Show notification using the current template
         /// </summary>
         /// <param name="duration">Displayed duration of the notification in ms (less or equal 0 means infinite duration)</param>
-        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-        public async Task ShowAsync(int duration = 0)
+        public void Show(int duration = 0)
         {
             Visibility = Visibility.Visible;
             VisualStateManager.GoToState(this, StateContentVisible, true);
 
             if (duration > 0)
             {
-                await Task.Delay(duration);
-                Dismiss();
+                _timer.Stop();
```

Comment:
You should move all this timer stop code to be first to ensure the timer doesn't fire between when you call the make visible state and stopping the timer.

Reply:
Even better.